src/Network/WebSockets/TSSocketManager.m

```diff
@@ -110,9 +128,10 @@ - (void)becomeActive
     }
 
     // Discard the old socket which is already closed or is closing.
-    [socket close];
-    self.status = kSocketStatusClosed;
-    socket.delegate = nil;
+    [self closeWebSocket];
```

Comment:
A number of properties are closely related and should always be updated in unison. I've DRY'ed that up with closeWebSocket.